rust/operator-binary/src/spark_k8s_controller.rs

```diff
@@ -208,31 +220,31 @@ fn pod_template(
                 "PYTHONPATH",
                 format!("$SPARK_HOME/python:{VOLUME_MOUNT_PATH_REQ}:$PYTHONPATH"),
             );
+        volumes.extend(vec![VolumeBuilder::new(VOLUME_MOUNT_NAME_REQ)
+            .empty_dir(EmptyDirVolumeSource::default())
+            .build()]);
     }
 
-    let mut template = PodBuilder::new();
-    template
-        .metadata_default()
-        .add_container(container.build())
-        .add_volumes(volumes);
-
     if let Some(container) = requirements_container.clone() {
-        template.add_init_container(container);
-        template.add_volume(
-            VolumeBuilder::new(VOLUME_MOUNT_NAME_REQ)
-                .empty_dir(EmptyDirVolumeSource::default())
-                .build(),
-        );
+        inits.get_or_insert_with(Vec::new).push(container);
     }
     if let Some(container) = job_container.clone() {
-        template.add_init_container(container);
-        template.add_volume(
-            VolumeBuilder::new(VOLUME_MOUNT_NAME_JOB)
-                .empty_dir(EmptyDirVolumeSource::default())
-                .build(),
-        );
+        inits.get_or_insert_with(Vec::new).push(container);
     }
-    template.build().context(PodTemplateSnafu)
+
+    let pod = Pod {
+        metadata: ObjectMeta::default(),
+        spec: Some(PodSpec {
+            containers: vec![container.build()],
+            init_containers: inits,
+            volumes: Some(volumes),
+            image_pull_secrets: spark_application.spark_image_pull_secrets(),
```

Comment:
Just a nitpick: this is now always set, unlike `image_pull_policy` which is set only if the user has configured it.

Reply:
Implemented as suggested.